.github/workflows/rename-dev-wheels.py

```diff
@@ -0,0 +1,9 @@
+import glob
+import os
+
+for filename in glob.glob('dist/*whl'):
+    # We remove the version number, dev number, and Git hash to ensure we have a
+    # predictable URL of the uploaded release asset that downstream projects can use.
+    pkg, _, remainder = filename.split('-', 2)
+    target = f'{pkg}-nightly-{remainder}'
+    os.rename(filename, target)
```

Comment:
I think it will be needed if we ever decide to test on different versions/platforms, which we might not *want*, but there is no guarantee that we can always support the same minimum Python across projects. For example, it seems @jokasimr just changed the `essreflectometry` minimum to 3.10?

Reply:
I can change it back to 3.8 if that is better.
I changed it because I thought there was an issue when I ran it with 3.8, but found out later that was not the case.

Reply:
You can revert if you like (I don't know what users depend on), but in principle, this is fine, we just need to make multiple versions here, but extending the matrix is trivial.